controllers/customer_controller.go

```diff
@@ -0,0 +1,203 @@
+			return r.handleDelete(req, instance)
+		}
+		// Finalizer doesn't exist so clean up is already done
+		return reconcilerUtil.DoNotRequeue()
+	}
+
+	// Add finalizer if it doesn't exist
+	if !finalizerUtil.HasFinalizer(instance, CustomerFinalizer) {
+		log.Info("Adding finalizer for instance" + req.Name)
+
+		finalizerUtil.AddFinalizer(instance, CustomerFinalizer)
+
+		err := r.Client.Update(context.TODO(), instance)
+		if err != nil {
+			return reconcilerUtil.ManageError(r.Client, instance, err, false)
+		}
+
+		return reconcilerUtil.DoNotRequeue()
+	}
+
+	return r.handleCreate(req, instance)
+}
+
+func (r *CustomerReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&jiraservicedeskv1alpha1.Customer{}).
+		Complete(r)
+}
+
+func (r *CustomerReconciler) handleCreate(req ctrl.Request, instance *jiraservicedeskv1alpha1.Customer) (ctrl.Result, error) {
+	log := r.Log.WithValues("customer", req.NamespacedName)
+
+	if len(instance.Status.AccountId) == 0 {
+		log.Info("Creating Jira Service Desk Customer: " + instance.Spec.DisplayName)
+
+		customer := r.JiraServiceDeskClient.GetCustomerFromCustomerCRForCreateCustomer(instance)
+		customerId, err := r.JiraServiceDeskClient.CreateCustomer(customer)
+		if err != nil {
+			return reconcilerUtil.ManageError(r.Client, instance, err, false)
+		}
+
+		instance.Status.AccountId = customerId
+		log.Info("Successfully created Jira Service Desk Customer: " + instance.Spec.DisplayName)
+	}
+
+	for _, projectKey := range instance.Spec.ProjectKeys {
+		if !contains(instance.Status.ProjectKeys, projectKey) {
+			err := r.JiraServiceDeskClient.AddCustomerToProject(instance.Status.AccountId, projectKey)
+			if err != nil {
+				return reconcilerUtil.ManageError(r.Client, instance, err, false)
+			}
+			instance.Status.ProjectKeys = append(instance.Status.ProjectKeys, projectKey)
+			log.Info("Successfully added Jira Service Desk Customer into project: " + projectKey)
+		}
+	}
+
+	for index, projectKey := range instance.Status.ProjectKeys {
+		if !contains(instance.Spec.ProjectKeys, projectKey) {
+			err := r.JiraServiceDeskClient.RemoveCustomerFromProject(instance.Status.AccountId, projectKey)
+			if err != nil {
+				return reconcilerUtil.ManageError(r.Client, instance, err, false)
+			}
+			instance.Status.ProjectKeys[index] = ""
+			log.Info("Successfully removed Jira Service Desk Customer from project: " + projectKey)
+		}
+	}
+
+	instance.Status.ProjectKeys = deleteEmpty(instance.Status.ProjectKeys)
+
+	return reconcilerUtil.ManageSuccess(r.Client, instance)
+}
+
+func (r *CustomerReconciler) handleDelete(req ctrl.Request, instance *jiraservicedeskv1alpha1.Customer) (ctrl.Result, error) {
+	log := r.Log.WithValues("customer", req.NamespacedName)
+
+	if instance == nil {
+		// Instance not found, nothing to do
+		return reconcilerUtil.DoNotRequeue()
+	}
+
+	log.Info("Removing Jira Service Desk Customer: " + instance.Spec.DisplayName)
```

Comment:
Removing project associations for jsd customer or something similar would be better suited here since we are not actually removing the customer